mcs/class/corlib/System.Runtime.CompilerServices/ConditionalWeakTable.cs

```diff
@@ -132,6 +133,17 @@ private void Rehash ()
 
 			uint newLength = (uint)HashHelpers.GetPrime (((int)(size / LOAD_FACTOR) << 1) | 1);
 
+			if (activeEnumeratorRefCount > 0)
+			{
+				// do not rehash if there are active enumerators
+				// Resize if there is no any free slot
+				if (size == data.Length)
+				{
+					Array.Resize<Ephemeron> (ref data, (int) newLength);
```

Comment:
You can't resize a hashtable without rehashing it.

Reply:
Additionally, Ephemeron arrays are special and must be registered with the runtime in order to produce the desired effect.